rust/processor/src/processors/account_transactions_processor.rs

```diff
@@ -2,86 +2,69 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use super::{ProcessingResult, ProcessorName, ProcessorTrait};
-use crate::{
-    models::account_transaction_models::account_transactions::AccountTransaction,
-    schema,
-    utils::database::{execute_in_chunks, get_config_table_chunk_size, PgDbPool},
-};
+use crate::{models::account_transaction_models::account_transactions::AccountTransaction, schema};
 use ahash::AHashMap;
 use anyhow::bail;
 use aptos_protos::transaction::v1::Transaction;
 use async_trait::async_trait;
-use diesel::{pg::Pg, query_builder::QueryFragment};
-use std::fmt::Debug;
 use tracing::error;
 
 pub struct AccountTransactionsProcessor {
-    connection_pool: PgDbPool,
-    per_table_chunk_sizes: AHashMap<String, usize>,
+    db_writer: crate::db_writer::DbWriter,
 }
 
-impl AccountTransactionsProcessor {
-    pub fn new(connection_pool: PgDbPool, per_table_chunk_sizes: AHashMap<String, usize>) -> Self {
-        Self {
-            connection_pool,
-            per_table_chunk_sizes,
-        }
-    }
-}
-
-impl Debug for AccountTransactionsProcessor {
+impl std::fmt::Debug for AccountTransactionsProcessor {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let state = &self.connection_pool.state();
+        let state = &self.connection_pool().state();
         write!(
             f,
-            "AccountTransactionsProcessor {{ connections: {:?}  idle_connections: {:?} }}",
-            state.connections, state.idle_connections
+            "{:} {{ connections: {:?}  idle_connections: {:?} }}",
+            self.name(),
+            state.connections,
+            state.idle_connections
         )
     }
 }
 
+impl AccountTransactionsProcessor {
+    pub fn new(db_writer: crate::db_writer::DbWriter) -> Self {
+        Self { db_writer }
+    }
+}
+
 async fn insert_to_db(
-    conn: PgDbPool,
+    db_writer: &crate::db_writer::DbWriter,
     name: &'static str,
     start_version: u64,
     end_version: u64,
-    account_transactions: &[AccountTransaction],
-    per_table_chunk_sizes: &AHashMap<String, usize>,
+    account_transactions: Vec<AccountTransaction>,
 ) -> Result<(), diesel::result::Error> {
     tracing::trace!(
         name = name,
         start_version = start_version,
         end_version = end_version,
         "Inserting to db",
     );
-    execute_in_chunks(
-        conn.clone(),
-        insert_account_transactions_query,
-        account_transactions,
-        get_config_table_chunk_size::<AccountTransaction>(
-            "account_transactions",
-            per_table_chunk_sizes,
-        ),
-    )
-    .await?;
+    db_writer
+        .send_in_chunks("account_transactions", account_transactions)
```

Comment:
would it be cleaner to have the `table_name` as a function in the `DBExecutable` struct? we don't have to worry about mismatching table_name and structs. 

or... even better if there's a way to get table_name from the `AccountTransaction` itself (i remember you were looking into that).

Reply:
yeah... I think this is where writing a macro or forking diesel might come into play